oci/table_oci_dns_zone.go

```diff
@@ -0,0 +1,281 @@
+				Description: "The authoritative nameservers for the zone.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getDnsZone,
+			},
+
+			// tags
+			{
+				Name:        "defined_tags",
+				Description: ColumnDescriptionDefinedTags,
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "freeform_tags",
+				Description: ColumnDescriptionFreefromTags,
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Standard Steampipe columns
+			{
+				Name:        "tags",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(dnsZoneTags),
+			},
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Name"),
+			},
+
+			// Standard OCI columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Id").Transform(ociRegionName),
+			},
+			{
+				Name:        "compartment_id",
+				Description: ColumnDescriptionCompartment,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CompartmentId"),
+			},
+			{
+				Name:        "tenant_id",
+				Description: ColumnDescriptionTenant,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getTenantId,
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listDnsZones(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("oci.listDnsZones", "Compartment", compartment, "OCI_REGION", region)
+
+	// Create Session
+	session, err := DNSService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	request := dns.ListZonesRequest{
+		CompartmentId: types.String(compartment),
+		RequestMetadata: oci_common.RequestMetadata{
+			RetryPolicy: getDefaultRetryPolicy(),
+		},
+	}
+
+	pagesLeft := true
+	for pagesLeft {
+		zones, err := session.DNSClient.ListZones(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, zone := range zones.Items {
+			d.StreamListItem(ctx, zone)
+		}
+		if zones.OpcNextPage != nil {
+			request.Page = zones.OpcNextPage
+		} else {
+			pagesLeft = false
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getDnsZone(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+	logger.Debug("oci.getDnsZone", "Compartment", compartment, "OCI_REGION", region)
+
+	// Rstrict the api call to only root compartment/ per region
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
```

Comment:
```> select name, id from oci_new.oci_dns_zone
+---------------+------------------------------------------------------+
| name          | id                                                   |
+---------------+------------------------------------------------------+
| test-comp.com | ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52 |
| turbot.com    | ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 |
| test-comp.com | ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52 |
| turbot.com    | ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 |
+---------------+------------------------------------------------------+
> select name, id, region  from oci_new.oci_dns_zone
+---------------+------------------------------------------------------+--------+
| name          | id                                                   | region |
+---------------+------------------------------------------------------+--------+
| test-comp.com | ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52 |        |
| turbot.com    | ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 |        |
| test-comp.com | ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52 |        |
| turbot.com    | ocid1.dns-zone.oc1..7ad8a63053bc467486483f39a0232889 |        |
+---------------+------------------------------------------------------+--------+
> select * from oci_new.oci_dns_zone where id = 'ocid1.dns-zone.oc1..eb09616ec01d4452833745aefece9b52'
Error: get call returned 2 results - the key column is not globally unique
> 

```
we are getting duplicate data in list call, get call is also not working and region column is always empty.
Seems like  this resource is same a cloudguard  service, but not sure, please check on this.

Reply:
updated getmatrix, removed region.